storagetransfer/api/StorageTransfer.Samples/QuickstartSample.cs

```diff
@@ -0,0 +1,59 @@
+﻿/**
+ * Copyright 2021 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+// [START storagetransfer_quickstart]
+using System;
+using Google.Cloud.StorageTransfer.V1;
+
+namespace StorageTransfer.Samples
+{
+    public class QuickstartSample
+    {
+        public TransferJob Quickstart(
+            // Your Google Cloud Project ID
+            string projectId = "my-project-id",
+            // The GCS bucket to transfer data from
+            string sourceBucket = "my-source-bucket",
+            // The GCS bucket to transfer data to
+            string sinkBucket = "my-sink-bucket")
+        {
+            TransferJob transferJob = new TransferJob
+            {
+                ProjectId = projectId,
+                TransferSpec = new TransferSpec
+                {
+                    GcsDataSink = new GcsData { BucketName = sourceBucket },
+                    GcsDataSource = new GcsData { BucketName = sinkBucket }
+                },
+                Status = TransferJob.Types.Status.Enabled
+            };
+
+            StorageTransferServiceClient client = StorageTransferServiceClient.Create();
+            TransferJob response = client.CreateTransferJob(new CreateTransferJobRequest { TransferJob = transferJob });
+            client.RunTransferJob(new RunTransferJobRequest
+            {
+                JobName = response.Name,
+                ProjectId = projectId
+            });

```

Comment:
I wonder if you really want to poll here, which shows users how to find out that their job has completed. You can then check `completedJob` for errors or for a result if the operation returns any.

```suggestion
            var runningJob = client.RunTransferJob(new RunTransferJobRequest
            {
                JobName = response.Name,
                ProjectId = projectId
            });
            
            var completedJob = runningJob.PollUntilCompleted();
```

Reply:
Interesting idea. The existing STS samples generally show transfer jobs as something you just schedule and then can poll for later (see the [samples page](https://cloud.google.com/storage-transfer/docs/create-manage-transfer-program#java)), I think because they can have schedules to run periodically and can take a long time. The docs also show you how to check the status of the latest operation associated with a job. So I think they're usually thought of as non-blocking. We didn't incorporate this into the quickstarts for the other languages, so I think I won't incorporate it here for now, but I can talk with the STS group and see what they think

Reply:
Sounds good, thanks.